lib/semantics/tools.cc

```diff
@@ -790,4 +790,16 @@ static Symbol &InstantiateSymbol(
   return result;
 }
 
+bool IsFunctionResultWithSameNameAsFunction(const Symbol &symbol) {
+  if ((symbol.has<semantics::ObjectEntityDetails>() &&
```

Comment:
Consider defining a function in tools.h (`IsFunctionResult(const Symbol &)`) and calling it here.  It might be a useful thing elsewhere and would hide some of the internal details.

Reply:
I agree with Peter.

Reply:
Done